File: src/data/catalog.ts
import type { CatalogDish } from '../types'

/**
 * Mocked master dish catalog — stands in for the shared taxonomy Swiggy/Zomato
 * maintain centrally (thousands of known dishes, pre-tagged with category,
 * type and tax slab) so a merchant rarely types a dish the platform hasn't
 * already seen. In production this would be a backend-owned, searchable
 * service (Elasticsearch/pgvector) instead of a bundled array.
 */
export const CATALOG: CatalogDish[] = [
  // Biryani & Rice
  { name: 'Chicken Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'non-veg', gst: 5, keywords: ['chicken', 'biryani'] },
  { name: 'Mutton Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'non-veg', gst: 5, keywords: ['mutton', 'biryani'] },
  { name: 'Egg Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'egg', gst: 5, keywords: ['egg', 'biryani'] },
  { name: 'Veg Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'veg', gst: 5, keywords: ['veg', 'biryani', 'vegetable'] },
  { name: 'Paneer Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'veg', gst: 5, keywords: ['paneer', 'biryani'] },
  { name: 'Fish Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'non-veg', gst: 5, keywords: ['fish', 'biryani'] },
  { name: 'Prawn Biryani', category: 'Biryani & Rice', subcategory: 'Biryani', type: 'non-veg', gst: 5, keywords: ['prawn', 'shrimp', 'biryani'] },
  { name: 'Jeera Rice', category: 'Biryani & Rice', subcategory: 'Rice', type: 'veg', gst: 5, keywords: ['jeera', 'rice', 'cumin'] },
  { name: 'Curd Rice', category: 'Biryani & Rice', subcategory: 'Rice', type: 'veg', gst: 5, keywords: ['curd', 'rice', 'yogurt'] },
  { name: 'Ghee Rice', category: 'Biryani & Rice', subcategory: 'Rice', type: 'veg', gst: 5, keywords: ['ghee', 'rice'] },

  // Starters
  { name: 'Chicken 65', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['chicken', '65'] },
  { name: 'Chilli Chicken', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['chilli', 'chicken'] },
  { name: 'Chicken Lollipop', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['chicken', 'lollipop'] },
  { name: 'Tandoori Chicken', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['tandoori', 'chicken'] },
  { name: 'Chicken Tikka', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['chicken', 'tikka'] },
  { name: 'Seekh Kebab', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['seekh', 'kebab', 'kabab'] },
  { name: 'Fish Fry', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['fish', 'fry'] },
  { name: 'Prawn Fry', category: 'Starters', subcategory: 'Non-Veg Starters', type: 'non-veg', gst: 5, keywords: ['prawn', 'fry', 'shrimp'] },
  { name: 'Paneer Tikka', category: 'Starters', subcategory: 'Veg Starters', type: 'veg', gst: 5, keywords: ['paneer', 'tikka'] },
  { name: 'Gobi Manchurian', category: 'Starters', subcategory: 'Veg Starters', type: 'veg', gst: 5, keywords: ['gobi', 'manchurian', 'cauliflower'] },
  { name: 'Veg Spring Roll', category: 'Starters', subcategory: 'Veg Starters', type: 'veg', gst: 5, keywords: ['spring', 'roll', 'veg'] },
  { name: 'Mushroom 65', category: 'Starters', subcategory: 'Veg Starters', type: 'veg', gst: 5, keywords: ['mushroom', '65'] },
  { name: 'Corn Chaat', category: 'Starters', subcategory: 'Veg Starters', type: 'veg', gst: 5, keywords: ['corn', 'chaat'] },
  { name: 'Papad', category: 'Starters', subcategory: 'Veg Starters', type: 'veg', gst: 5, keywords: ['papad', 'papadum'] },

  // Main Course
  { name: 'Butter Chicken', category: 'Main Course', subcategory: 'Non-Veg Curries', type: 'non-veg', gst: 5, keywords: ['butter', 'chicken', 'makhani'] },
  { name: 'Chicken Curry', category: 'Main Course', subcategory: 'Non-Veg Curries', type: 'non-veg', gst: 5, keywords: ['chicken', 'curry'] },
  { name: 'Mutton Curry', category: 'Main Course', subcategory: 'Non-Veg Curries', type: 'non-veg', gst: 5, keywords: ['mutton', 'curry'] },
  { name: 'Chicken Chettinad', category: 'Main Course', subcategory: 'Non-Veg Curries', type: 'non-veg', gst: 5, keywords: ['chicken', 'chettinad'] },
  { name: 'Fish Curry', category: 'Main Course', subcategory: 'Non-Veg Curries', type: 'non-veg', gst: 5, keywords: ['fish', 'curry'] },
  { name: 'Egg Curry', category: 'Main Course', subcategory: 'Non-Veg Curries', type: 'egg', gst: 5, keywords: ['egg', 'curry'] },
  { name: 'Paneer Butter Masala', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['paneer', 'butter', 'masala'] },
  { name: 'Paneer Tikka Masala', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['paneer', 'tikka', 'masala'] },
  { name: 'Dal Tadka', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['dal', 'tadka', 'lentil'] },
  { name: 'Dal Makhani', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['dal', 'makhani', 'lentil'] },
  { name: 'Chana Masala', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['chana', 'chole', 'masala', 'chickpea'] },
  { name: 'Palak Paneer', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['palak', 'paneer', 'spinach'] },
  { name: 'Mixed Veg Curry', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['mixed', 'veg', 'curry', 'vegetable'] },
  { name: 'Malai Kofta', category: 'Main Course', subcategory: 'Veg Curries', type: 'veg', gst: 5, keywords: ['malai', 'kofta'] },

  // Breads
  { name: 'Butter Naan', category: 'Breads', subcategory: 'Naan', type: 'veg', gst: 5, keywords: ['naan', 'butter'] },
  { name: 'Garlic Naan', category: 'Breads', subcategory: 'Naan', type: 'veg', gst: 5, keywords: ['naan', 'garlic'] },
  { name: 'Tandoori Roti', category: 'Breads', subcategory: 'Roti', type: 'veg', gst: 5, keywords: ['roti', 'tandoori'] },
  { name: 'Lachha Paratha', category: 'Breads', subcategory: 'Paratha', type: 'veg', gst: 5, keywords: ['paratha', 'lachha'] },
  { name: 'Plain Paratha', category: 'Breads', subcategory: 'Paratha', type: 'veg', gst: 5, keywords: ['paratha', 'plain'] },

  // South Indian
  { name: 'Masala Dosa', category: 'South Indian', subcategory: 'Dosa', type: 'veg', gst: 5, keywords: ['dosa', 'masala'] },
  { name: 'Plain Dosa', category: 'South Indian', subcategory: 'Dosa', type: 'veg', gst: 5, keywords: ['dosa', 'plain'] },
  { name: 'Rava Dosa', category: 'South Indian', subcategory: 'Dosa', type: 'veg', gst: 5, keywords: ['dosa', 'rava'] },
  { name: 'Idli', category: 'South Indian', subcategory: 'Idli & Vada', type: 'veg', gst: 5, keywords: ['idli'] },
  { name: 'Medu Vada', category: 'South Indian', subcategory: 'Idli & Vada', type: 'veg', gst: 5, keywords: ['vada', 'medu'] },
  { name: 'Uttapam', category: 'South Indian', subcategory: 'Dosa', type: 'veg', gst: 5, keywords: ['uttapam'] },
  { name: 'Pongal', category: 'South Indian', subcategory: 'Rice', type: 'veg', gst: 5, keywords: ['pongal'] },

  // Chinese
  { name: 'Veg Fried Rice', category: 'Chinese', subcategory: 'Rice & Noodles', type: 'veg', gst: 5, keywords: ['fried', 'rice', 'veg'] },
  { name: 'Chicken Fried Rice', category: 'Chinese', subcategory: 'Rice & Noodles', type: 'non-veg', gst: 5, keywords: ['fried', 'rice', 'chicken'] },
  { name: 'Veg Noodles', category: 'Chinese', subcategory: 'Rice & Noodles', type: 'veg', gst: 5, keywords: ['noodles', 'veg', 'hakka'] },
  { name: 'Chicken Noodles', category: 'Chinese', subcategory: 'Rice & Noodles', type: 'non-veg', gst: 5, keywords: ['noodles', 'chicken', 'hakka'] },
  { name: 'Veg Manchurian', category: 'Chinese', subcategory: 'Gravy', type: 'veg', gst: 5, keywords: ['manchurian', 'veg'] },
  { name: 'Chicken Manchurian', category: 'Chinese', subcategory: 'Gravy', type: 'non-veg', gst: 5, keywords: ['manchurian', 'chicken'] },

  // Desserts
  { name: 'Gulab Jamun', category: 'Desserts', subcategory: 'Indian Sweets', type: 'veg', gst: 5, keywords: ['gulab', 'jamun'] },
  { name: 'Rasmalai', category: 'Desserts', subcategory: 'Indian Sweets', type: 'veg', gst: 5, keywords: ['rasmalai'] },
  { name: 'Kulfi', category: 'Desserts', subcategory: 'Ice Cream', type: 'veg', gst: 5, keywords: ['kulfi'] },
  { name: 'Ice Cream', category: 'Desserts', subcategory: 'Ice Cream', type: 'veg', gst: 5, keywords: ['ice', 'cream'] },
  { name: 'Double Ka Meetha', category: 'Desserts', subcategory: 'Indian Sweets', type: 'veg', gst: 5, keywords: ['double', 'ka', 'meetha'] },

  // Beverages
  { name: 'Sweet Lassi', category: 'Beverages', subcategory: 'Lassi & Shakes', type: 'veg', gst: 5, keywords: ['lassi', 'sweet'] },
  { name: 'Masala Chaas', category: 'Beverages', subcategory: 'Lassi & Shakes', type: 'veg', gst: 5, keywords: ['chaas', 'buttermilk', 'masala'] },
  { name: 'Mango Shake', category: 'Beverages', subcategory: 'Lassi & Shakes', type: 'veg', gst: 5, keywords: ['mango', 'shake'] },
  { name: 'Fresh Lime Soda', category: 'Beverages', subcategory: 'Soft Drinks', type: 'veg', gst: 5, keywords: ['lime', 'soda', 'nimbu'] },
  { name: 'Masala Chai', category: 'Beverages', subcategory: 'Tea & Coffee', type: 'veg', gst: 5, keywords: ['chai', 'tea', 'masala'] },
  { name: 'Filter Coffee', category: 'Beverages', subcategory: 'Tea & Coffee', type: 'veg', gst: 5, keywords: ['coffee', 'filter'] },
  { name: 'Cold Coffee', category: 'Beverages', subcategory: 'Tea & Coffee', type: 'veg', gst: 5, keywords: ['cold', 'coffee'] },
  { name: 'Iced Tea', category: 'Beverages', subcategory: 'Tea & Coffee', type: 'veg', gst: 5, keywords: ['iced', 'tea'] },
  { name: 'Soft Drink', category: 'Beverages', subcategory: 'Soft Drinks', type: 'veg', gst: 5, keywords: ['soft', 'drink', 'coke', 'pepsi', 'sprite'] },
  { name: 'Buttermilk', category: 'Beverages', subcategory: 'Lassi & Shakes', type: 'veg', gst: 5, keywords: ['buttermilk', 'majjige'] },

  // Breakfast
  { name: 'Poha', category: 'Breakfast', subcategory: 'North Indian Breakfast', type: 'veg', gst: 5, keywords: ['poha', 'pohe', 'flattened', 'rice'] },
  { name: 'Upma', category: 'Breakfast', subcategory: 'South Indian Breakfast', type: 'veg', gst: 5, keywords: ['upma', 'uppma', 'rava'] },
  { name: 'Aloo Paratha', category: 'Breakfast', subcategory: 'North Indian Breakfast', type: 'veg', gst: 5, keywords: ['aloo', 'paratha', 'potato'] },
  { name: 'Chole Bhature', category: 'Breakfast', subcategory: 'North Indian Breakfast', type: 'veg', gst: 5, keywords: ['chole', 'bhature', 'chana', 'bhatura'] },
  { name: 'Vada Pav', category: 'Breakfast', subcategory: 'Street Food', type: 'veg', gst: 5, keywords: ['vada', 'pav'] },
  { name: 'Misal Pav', category: 'Breakfast', subcategory: 'Street Food', type: 'veg', gst: 5, keywords: ['misal', 'pav'] },
  { name: 'Masala Omelette', category: 'Breakfast', subcategory: 'Egg Breakfast', type: 'egg', gst: 5, keywords: ['omelette', 'omelet', 'masala', 'egg'] },
  { name: 'Boiled Eggs', category: 'Breakfast', subcategory: 'Egg Breakfast', type: 'egg', gst: 5, keywords: ['boiled', 'egg', 'eggs'] },
  { name: 'Bread Butter Jam', category: 'Breakfast', subcategory: 'Continental Breakfast', type: 'veg', gst: 5, keywords: ['bread', 'butter', 'jam', 'toast'] },
  { name: 'Cornflakes with Milk', category: 'Breakfast', subcategory: 'Continental Breakfast', type: 'veg', gst: 5, keywords: ['cornflakes', 'cereal', 'milk'] },

  // Fast Food & Continental
  { name: 'Veg Burger', category: 'Fast Food', subcategory: 'Burgers', type: 'veg', gst: 5, keywords: ['veg', 'burger'] },
  { name: 'Chicken Burger', category: 'Fast Food', subcategory: 'Burgers', type: 'non-veg', gst: 5, keywords: ['chicken', 'burger'] },
  { name: 'Paneer Burger', category: 'Fast Food', subcategory: 'Burgers', type: 'veg', gst: 5, keywords: ['paneer', 'burger'] },
  { name: 'Veg Pizza', category: 'Fast Food', subcategory: 'Pizza', type: 'veg', gst: 5, keywords: ['veg', 'pizza', 'margherita'] },
  { name: 'Chicken Pizza', category: 'Fast Food', subcategory: 'Pizza', type: 'non-veg', gst: 5, keywords: ['chicken', 'pizza'] },
  { name: 'Cheese Pizza', category: 'Fast Food', subcategory: 'Pizza', type: 'veg', gst: 5, keywords: ['cheese', 'pizza'] },
  { name: 'Veg Sandwich', category: 'Fast Food', subcategory: 'Sandwiches', type: 'veg', gst: 5, keywords: ['veg', 'sandwich'] },
  { name: 'Grilled Chicken Sandwich', category: 'Fast Food', subcategory: 'Sandwiches', type: 'non-veg', gst: 5, keywords: ['grilled', 'chicken', 'sandwich'] },
  { name: 'Club Sandwich', category: 'Fast Food', subcategory: 'Sandwiches', type: 'veg', gst: 5, keywords: ['club', 'sandwich'] },
  { name: 'French Fries', category: 'Fast Food', subcategory: 'Sides', type: 'veg', gst: 5, keywords: ['french', 'fries', 'fry'] },
  { name: 'Pasta Alfredo', category: 'Fast Food', subcategory: 'Pasta', type: 'veg', gst: 5, keywords: ['pasta', 'alfredo', 'white', 'sauce'] },
  { name: 'Pasta Arrabbiata', category: 'Fast Food', subcategory: 'Pasta', type: 'veg', gst: 5, keywords: ['pasta', 'arrabbiata', 'red', 'sauce'] },
  { name: 'Veg Momos', category: 'Fast Food', subcategory: 'Momos', type: 'veg', gst: 5, keywords: ['momo', 'momos', 'veg', 'dumpling'] },
  { name: 'Chicken Momos', category: 'Fast Food', subcategory: 'Momos', type: 'non-veg', gst: 5, keywords: ['momo', 'momos', 'chicken', 'dumpling'] },
  { name: 'Chicken Shawarma Roll', category: 'Fast Food', subcategory: 'Rolls', type: 'non-veg', gst: 5, keywords: ['shawarma', 'chicken', 'roll'] },
  { name: 'Egg Roll', category: 'Fast Food', subcategory: 'Rolls', type: 'egg', gst: 5, keywords: ['egg', 'roll', 'kathi'] },
  { name: 'Veg Frankie', category: 'Fast Food', subcategory: 'Rolls', type: 'veg', gst: 5, keywords: ['veg', 'frankie', 'roll', 'kathi'] },
]

/** Category → default GST slab, used when a typed item has no catalog match. */
export const CATEGORY_DEFAULT_GST: Record<string, number> = {
  'Biryani & Rice': 5,
  Starters: 5,
  'Main Course': 5,
  Breads: 5,
  'South Indian': 5,
  Chinese: 5,
  Desserts: 5,
  Beverages: 5,
  Breakfast: 5,
  'Fast Food': 5,
}

export const CATEGORIES = Array.from(new Set(CATALOG.map((d) => d.category)))

export function subcategoriesFor(category: string): string[] {
  return Array.from(new Set(CATALOG.filter((d) => d.category === category).map((d) => d.subcategory)))
}

File: backend/app/catalog.py
"""Master dish catalog — Python port of ``src/data/catalog.ts``.

Stands in for the shared taxonomy Swiggy/Zomato maintain centrally
(thousands of known dishes, pre-tagged with category, type and tax slab)
so a merchant rarely types a dish the platform hasn't already seen. In a
real deployment this table lives in Postgres (or an Elasticsearch/pgvector
index once the catalog grows past a few thousand rows) — it is a plain
Python list here purely so the matching logic in ``matching.py`` has no
infrastructure dependency and can be unit tested in isolation.
"""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(frozen=True)
class CatalogDish:
    name: str
    category: str
    subcategory: str
    type: str  # "veg" | "non-veg" | "egg"
    gst: float
    keywords: tuple[str, ...] = field(default_factory=tuple)


def _d(name: str, category: str, subcategory: str, type_: str, gst: float, keywords: list[str]) -> CatalogDish:
    return CatalogDish(name, category, subcategory, type_, gst, tuple(keywords))


CATALOG: list[CatalogDish] = [
    # Biryani & Rice
    _d("Chicken Biryani", "Biryani & Rice", "Biryani", "non-veg", 5, ["chicken", "biryani"]),
    _d("Mutton Biryani", "Biryani & Rice", "Biryani", "non-veg", 5, ["mutton", "biryani"]),
    _d("Egg Biryani", "Biryani & Rice", "Biryani", "egg", 5, ["egg", "biryani"]),
    _d("Veg Biryani", "Biryani & Rice", "Biryani", "veg", 5, ["veg", "biryani", "vegetable"]),
    _d("Paneer Biryani", "Biryani & Rice", "Biryani", "veg", 5, ["paneer", "biryani"]),
    _d("Fish Biryani", "Biryani & Rice", "Biryani", "non-veg", 5, ["fish", "biryani"]),
    _d("Prawn Biryani", "Biryani & Rice", "Biryani", "non-veg", 5, ["prawn", "shrimp", "biryani"]),
    _d("Jeera Rice", "Biryani & Rice", "Rice", "veg", 5, ["jeera", "rice", "cumin"]),
    _d("Curd Rice", "Biryani & Rice", "Rice", "veg", 5, ["curd", "rice", "yogurt"]),
    _d("Ghee Rice", "Biryani & Rice", "Rice", "veg", 5, ["ghee", "rice"]),
    # Starters
    _d("Chicken 65", "Starters", "Non-Veg Starters", "non-veg", 5, ["chicken", "65"]),
    _d("Chilli Chicken", "Starters", "Non-Veg Starters", "non-veg", 5, ["chilli", "chicken"]),
    _d("Chicken Lollipop", "Starters", "Non-Veg Starters", "non-veg", 5, ["chicken", "lollipop"]),
    _d("Tandoori Chicken", "Starters", "Non-Veg Starters", "non-veg", 5, ["tandoori", "chicken"]),
    _d("Chicken Tikka", "Starters", "Non-Veg Starters", "non-veg", 5, ["chicken", "tikka"]),
    _d("Seekh Kebab", "Starters", "Non-Veg Starters", "non-veg", 5, ["seekh", "kebab", "kabab"]),
    _d("Fish Fry", "Starters", "Non-Veg Starters", "non-veg", 5, ["fish", "fry"]),
    _d("Prawn Fry", "Starters", "Non-Veg Starters", "non-veg", 5, ["prawn", "fry", "shrimp"]),
    _d("Paneer Tikka", "Starters", "Veg Starters", "veg", 5, ["paneer", "tikka"]),
    _d("Gobi Manchurian", "Starters", "Veg Starters", "veg", 5, ["gobi", "manchurian", "cauliflower"]),
    _d("Veg Spring Roll", "Starters", "Veg Starters", "veg", 5, ["spring", "roll", "veg"]),
    _d("Mushroom 65", "Starters", "Veg Starters", "veg", 5, ["mushroom", "65"]),
    _d("Corn Chaat", "Starters", "Veg Starters", "veg", 5, ["corn", "chaat"]),
    _d("Papad", "Starters", "Veg Starters", "veg", 5, ["papad", "papadum"]),
    # Main Course
    _d("Butter Chicken", "Main Course", "Non-Veg Curries", "non-veg", 5, ["butter", "chicken", "makhani"]),
    _d("Chicken Curry", "Main Course", "Non-Veg Curries", "non-veg", 5, ["chicken", "curry"]),
    _d("Mutton Curry", "Main Course", "Non-Veg Curries", "non-veg", 5, ["mutton", "curry"]),
    _d("Chicken Chettinad", "Main Course", "Non-Veg Curries", "non-veg", 5, ["chicken", "chettinad"]),
    _d("Fish Curry", "Main Course", "Non-Veg Curries", "non-veg", 5, ["fish", "curry"]),
    _d("Egg Curry", "Main Course", "Non-Veg Curries", "egg", 5, ["egg", "curry"]),
    _d("Paneer Butter Masala", "Main Course", "Veg Curries", "veg", 5, ["paneer", "butter", "masala"]),
    _d("Paneer Tikka Masala", "Main Course", "Veg Curries", "veg", 5, ["paneer", "tikka", "masala"]),
    _d("Dal Tadka", "Main Course", "Veg Curries", "veg", 5, ["dal", "tadka", "lentil"]),
    _d("Dal Makhani", "Main Course", "Veg Curries", "veg", 5, ["dal", "makhani", "lentil"]),
    _d("Chana Masala", "Main Course", "Veg Curries", "veg", 5, ["chana", "chole", "masala", "chickpea"]),
    _d("Palak Paneer", "Main Course", "Veg Curries", "veg", 5, ["palak", "paneer", "spinach"]),
    _d("Mixed Veg Curry", "Main Course", "Veg Curries", "veg", 5, ["mixed", "veg", "curry", "vegetable"]),
    _d("Malai Kofta", "Main Course", "Veg Curries", "veg", 5, ["malai", "kofta"]),
    # Breads
    _d("Butter Naan", "Breads", "Naan", "veg", 5, ["naan", "butter"]),
    _d("Garlic Naan", "Breads", "Naan", "veg", 5, ["naan", "garlic"]),
    _d("Tandoori Roti", "Breads", "Roti", "veg", 5, ["roti", "tandoori"]),
    _d("Lachha Paratha", "Breads", "Paratha", "veg", 5, ["paratha", "lachha"]),
    _d("Plain Paratha", "Breads", "Paratha", "veg", 5, ["paratha", "plain"]),
    # South Indian
    _d("Masala Dosa", "South Indian", "Dosa", "veg", 5, ["dosa", "masala"]),
    _d("Plain Dosa", "South Indian", "Dosa", "veg", 5, ["dosa", "plain"]),
    _d("Rava Dosa", "South Indian", "Dosa", "veg", 5, ["dosa", "rava"]),
    _d("Idli", "South Indian", "Idli & Vada", "veg", 5, ["idli"]),
    _d("Medu Vada", "South Indian", "Idli & Vada", "veg", 5, ["vada", "medu"]),
    _d("Uttapam", "South Indian", "Dosa", "veg", 5, ["uttapam"]),
    _d("Pongal", "South Indian", "Rice", "veg", 5, ["pongal"]),
    # Chinese
    _d("Veg Fried Rice", "Chinese", "Rice & Noodles", "veg", 5, ["fried", "rice", "veg"]),
    _d("Chicken Fried Rice", "Chinese", "Rice & Noodles", "non-veg", 5, ["fried", "rice", "chicken"]),
    _d("Veg Noodles", "Chinese", "Rice & Noodles", "veg", 5, ["noodles", "veg", "hakka"]),
    _d("Chicken Noodles", "Chinese", "Rice & Noodles", "non-veg", 5, ["noodles", "chicken", "hakka"]),
    _d("Veg Manchurian", "Chinese", "Gravy", "veg", 5, ["manchurian", "veg"]),
    _d("Chicken Manchurian", "Chinese", "Gravy", "non-veg", 5, ["manchurian", "chicken"]),
    # Desserts
    _d("Gulab Jamun", "Desserts", "Indian Sweets", "veg", 5, ["gulab", "jamun"]),
    _d("Rasmalai", "Desserts", "Indian Sweets", "veg", 5, ["rasmalai"]),
    _d("Kulfi", "Desserts", "Ice Cream", "veg", 5, ["kulfi"]),
    _d("Ice Cream", "Desserts", "Ice Cream", "veg", 5, ["ice", "cream"]),
    _d("Double Ka Meetha", "Desserts", "Indian Sweets", "veg", 5, ["double", "ka", "meetha"]),
    # Beverages
    _d("Sweet Lassi", "Beverages", "Lassi & Shakes", "veg", 5, ["lassi", "sweet"]),
    _d("Masala Chaas", "Beverages", "Lassi & Shakes", "veg", 5, ["chaas", "buttermilk", "masala"]),
    _d("Mango Shake", "Beverages", "Lassi & Shakes", "veg", 5, ["mango", "shake"]),
    _d("Fresh Lime Soda", "Beverages", "Soft Drinks", "veg", 5, ["lime", "soda", "nimbu"]),
    _d("Masala Chai", "Beverages", "Tea & Coffee", "veg", 5, ["chai", "tea", "masala"]),
    _d("Filter Coffee", "Beverages", "Tea & Coffee", "veg", 5, ["coffee", "filter"]),
    _d("Cold Coffee", "Beverages", "Tea & Coffee", "veg", 5, ["cold", "coffee"]),
    _d("Iced Tea", "Beverages", "Tea & Coffee", "veg", 5, ["iced", "tea"]),
    _d("Soft Drink", "Beverages", "Soft Drinks", "veg", 5, ["soft", "drink", "coke", "pepsi", "sprite"]),
    _d("Buttermilk", "Beverages", "Lassi & Shakes", "veg", 5, ["buttermilk", "majjige"]),
    # Breakfast
    _d("Poha", "Breakfast", "North Indian Breakfast", "veg", 5, ["poha", "pohe", "flattened", "rice"]),
    _d("Upma", "Breakfast", "South Indian Breakfast", "veg", 5, ["upma", "uppma", "rava"]),
    _d("Aloo Paratha", "Breakfast", "North Indian Breakfast", "veg", 5, ["aloo", "paratha", "potato"]),
    _d("Chole Bhature", "Breakfast", "North Indian Breakfast", "veg", 5, ["chole", "bhature", "chana", "bhatura"]),
    _d("Vada Pav", "Breakfast", "Street Food", "veg", 5, ["vada", "pav"]),
    _d("Misal Pav", "Breakfast", "Street Food", "veg", 5, ["misal", "pav"]),
    _d("Masala Omelette", "Breakfast", "Egg Breakfast", "egg", 5, ["omelette", "omelet", "masala", "egg"]),
    _d("Boiled Eggs", "Breakfast", "Egg Breakfast", "egg", 5, ["boiled", "egg", "eggs"]),
    _d("Bread Butter Jam", "Breakfast", "Continental Breakfast", "veg", 5, ["bread", "butter", "jam", "toast"]),
    _d("Cornflakes with Milk", "Breakfast", "Continental Breakfast", "veg", 5, ["cornflakes", "cereal", "milk"]),
    # Fast Food & Continental
    _d("Veg Burger", "Fast Food", "Burgers", "veg", 5, ["veg", "burger"]),
    _d("Chicken Burger", "Fast Food", "Burgers", "non-veg", 5, ["chicken", "burger"]),
    _d("Paneer Burger", "Fast Food", "Burgers", "veg", 5, ["paneer", "burger"]),
    _d("Veg Pizza", "Fast Food", "Pizza", "veg", 5, ["veg", "pizza", "margherita"]),
    _d("Chicken Pizza", "Fast Food", "Pizza", "non-veg", 5, ["chicken", "pizza"]),
    _d("Cheese Pizza", "Fast Food", "Pizza", "veg", 5, ["cheese", "pizza"]),
    _d("Veg Sandwich", "Fast Food", "Sandwiches", "veg", 5, ["veg", "sandwich"]),
    _d("Grilled Chicken Sandwich", "Fast Food", "Sandwiches", "non-veg", 5, ["grilled", "chicken", "sandwich"]),
    _d("Club Sandwich", "Fast Food", "Sandwiches", "veg", 5, ["club", "sandwich"]),
    _d("French Fries", "Fast Food", "Sides", "veg", 5, ["french", "fries", "fry"]),
    _d("Pasta Alfredo", "Fast Food", "Pasta", "veg", 5, ["pasta", "alfredo", "white", "sauce"]),
    _d("Pasta Arrabbiata", "Fast Food", "Pasta", "veg", 5, ["pasta", "arrabbiata", "red", "sauce"]),
    _d("Veg Momos", "Fast Food", "Momos", "veg", 5, ["momo", "momos", "veg", "dumpling"]),
    _d("Chicken Momos", "Fast Food", "Momos", "non-veg", 5, ["momo", "momos", "chicken", "dumpling"]),
    _d("Chicken Shawarma Roll", "Fast Food", "Rolls", "non-veg", 5, ["shawarma", "chicken", "roll"]),
    _d("Egg Roll", "Fast Food", "Rolls", "egg", 5, ["egg", "roll", "kathi"]),
    _d("Veg Frankie", "Fast Food", "Rolls", "veg", 5, ["veg", "frankie", "roll", "kathi"]),
]

# Category -> default GST slab, used when a typed item has no catalog match.
CATEGORY_DEFAULT_GST: dict[str, float] = {
    "Biryani & Rice": 5,
    "Starters": 5,
    "Main Course": 5,
    "Breads": 5,
    "South Indian": 5,
    "Chinese": 5,
    "Desserts": 5,
    "Beverages": 5,
    "Breakfast": 5,
    "Fast Food": 5,
}

CATEGORIES: list[str] = list(dict.fromkeys(d.category for d in CATALOG))


def subcategories_for(category: str) -> list[str]:
    return list(dict.fromkeys(d.subcategory for d in CATALOG if d.category == category))
